src/oscar/apps/offer/abstract_models.py

```diff
@@ -1119,10 +1119,11 @@ def extract_ids(self):
         """
         Extract all SKU- or UPC-like strings from the file
         """
-        for line in open(self.filepath, 'r'):
-            for id in re.split('[^\w:\.-]', line):
-                if id:
-                    yield id
+        with open(self.filepath, 'r') as fh:
+            for line in fh.readlines():
```

Comment:
`readlines()` loads the entire contents of the file into memory. I think it's better to just loop over the file object itself: `for line in fh` which is memory-efficient.